core/services/relay/evm/mercury/wsrpc/metrics.go

```diff
@@ -0,0 +1,48 @@
+package wsrpc
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type reqStatus string
+
+const (
+	statusSuccess reqStatus = "success"
+	statusFailed  reqStatus = "failed"
+)
+
+var (
+	aliveMetric = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: "mercury",
+		Name:      "wsrpc_connection_alive",
+		Help:      "Total time spent connected to the Mercury WSRPC server",
+	})
+	requestsStatusMetric = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "mercury",
+		Name:      "wsrpc_requests_status_count",
+		Help:      "Number of request status made to the Mercury WSRPC server",
+	}, []string{"status"})
+
+	requestLatencyMetric = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Should this instead be a histogram?

Reply:
See https://prometheus.io/docs/concepts/metric_types/ - expect the gauge will give us the last latency value instead of showing us all of the observations

Reply:
This is true. The gauge will limit us to whatever scrape interval we use, all other reqs will be discarded.
Seeing as Transmit will be pretty high-frequency, its probably better to use a histogram